tools/documentation_updater/documentation_updater.py

```diff
@@ -0,0 +1,214 @@
+import argparse
+import logging
+import os
+import re
+import ruamel.yaml
+from ruamel.yaml.scalarstring import FoldedScalarString
+from sys import exit
+
+REGIMES = [
+    'check',
+    'update'
+    ]
+
+REGIME = ''
+
+LOG_LEVELS = {
+    'critical': logging.CRITICAL,
+    'error': logging.ERROR,
+    'warning': logging.WARNING,
+    'info': logging.INFO,
+    'debug': logging.DEBUG
+}
+
+
+def parse():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-r', '--readme-dir', type=str, default='models',
+                        help='Path to root directory with models descriptions')
+    parser.add_argument('-c', '--config-dir', type=str,
+                        help='Path to root directory with topologies configs '
+                             '(by default used directory from "--readme-dir" key')
+
+    parser.add_argument('--deprecated_representation', type=bool, default=False,
+                        help="Used for old topology's representation")
+    parser.add_argument('-l', '--list', type=str, default='list_topologies.yml',
+                        help='DEPRECATED: file with topologies list')
+    parser.add_argument('-o', '--out-file', type=str,
+                        help='DEPRECATED: output file with topologies list '
+                             '(by default used original file from --out-file key)')
+
+    parser.add_argument('--regime', type=str, choices=REGIMES, default='check',
+                        help='Script work regime: "check" only finds diffs, "update" - updates values')
+    parser.add_argument('--log-level', choices=LOG_LEVELS.keys(), default='warning',
+                        help='Level of logging')
+    return parser.parse_args()
+
+
+def collect_readme(directory, ignored_files=('index.md',)):
+    if not os.path.isdir(directory):
+        logging.critical("Directory {} does not exist. Please check '--readme-dir' option.".format(os.path.abspath(directory)))
+        exit(1)
+    files = {}
+    for r, d, f in os.walk(directory):
+        for file in f:
+            if '.md' not in file or file in ignored_files:
+                continue
+            files[file.replace(".md", "")] = os.path.join(r, file)
+    logging.info('Collected {} readme files'.format(len(files)))
+    if not files:
+        logging.error("No markdown file found in {}. Exceptions - {}. Ensure, that you set right directory.".format(directory, ignored_files))
+        exit(1)
+    return files
+
+
+def convert(lines):
+    def flatten_links(s):
+        links = re.findall(r"\[.*?\]\(.*?\)", s)
+        for link in links:
+            plain_link = re.sub('[\[\]]', '', link.replace('](', ' (').replace('(', '<').replace(')', '>'))
+            s = s.replace(link, plain_link)
+        return s
```

Comment:
Seems like all of this could be done with a single replacement operation, something like `re.sub(r'\[(.*?\)]\((.*?)\)', r'\1 <\2>', s)`.